data/transactions/logic/eval.go

```diff
@@ -685,9 +792,30 @@ func boolToUint(x bool) uint64 {
 	return 0
 }
 
-// MaxStackDepth should move to consensus params
+// MaxStackDepth should not change unless gated by a teal version change / consensus upgrade.
 const MaxStackDepth = 1000
 
+func (cx *EvalContext) remainingBudget() int {
```

Comment:
might be safer to make 0 the lower bound of this method, but I don't see any logical flaw here.